src/core/url_store/atoms/urlStore.ts

```diff
@@ -75,9 +76,9 @@ export const urlStoreAtom = createAtom(
         if (initialState.map) {
           actions.push(
             currentMapPositionAtom.setCurrentMapPosition({
-              zoom: Number(initialState.map[0]),
-              lng: Number(initialState.map[1]),
-              lat: Number(initialState.map[2]),
+              zoom: Number(initialState.map[0]) - URL_ZOOM_OFFSET,
```

Comment:
Changes in the structure of the external url should not affect the application's internal state. these are different levels of abstraction. Do it in URLDataInSearchEncoder